SmartPlaces.Facilities/lib/IngestionManager/src/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,35 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="ServiceCollectionExtensions.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Microsoft.SmartPlaces.Facilities.IngestionManager.Extensions
+{
+    using System.Reflection;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.SmartPlaces.Facilities.IngestionManager;
+    using Microsoft.SmartPlaces.Facilities.IngestionManager.AzureDigitalTwins;
+    using Microsoft.SmartPlaces.Facilities.IngestionManager.Interfaces;
+
+    public static class ServiceCollectionExtensions
+    {
+        public static IServiceCollection AddIngestionManager<TOptions>(this IServiceCollection services, Action<TOptions> options)
+            where TOptions : IngestionManagerOptions
+        {
+            services.AddOptions<TOptions>()
+                    .Configure(options)
+                    .ValidateDataAnnotations()
+                    .ValidateOnStart();
+
+            services.AddSingleton<IOutputGraphManager, AdtGraphManager<TOptions>>();
+
+            services.AddHttpClient("Microsoft.SmartPlaces.Facilities", options =>
+            {
+                options.DefaultRequestHeaders.Add("ms-smartfacilities-version", Assembly.GetExecutingAssembly().GetName().Version?.ToString() ?? "0.0.0");
```

Comment:
Is there anything more useful we could leave here besides Assembly version? OpenSourcing makes the value less ..valuable. The header could still be used to semi-estimate adoption, but the exact version less so.